queries.yaml

```diff
@@ -972,6 +972,45 @@ post_process:
       items_matching:
         kind: station
       rank_key: kind_tile_rank
+  - fn: vectordatasource.transform.keep_n_features_gridded
+    params:
+      source_layer: places
+      start_zoom: 8
+      end_zoom: 9
+      items_matching: { kind: locality }
+      max_items: 1
+      grid_width: 3
+      sorting_keys:
+        - { sort_key: 'min_zoom', reverse: False }
+        - { sort_key: 'collision_rank', reverse: False }
+        - { sort_key: 'population', reverse: True }
+        - { sort_key: 'id', reverse: True }
+  - fn: vectordatasource.transform.keep_n_features_gridded
+    params:
+      source_layer: places
+      start_zoom: 9
+      end_zoom: 11
+      items_matching: { kind: locality }
+      max_items: 1
+      grid_width: 6
+      sorting_keys:
+        - { sort_key: 'min_zoom', reverse: False }
+        - { sort_key: 'collision_rank', reverse: False }
+        - { sort_key: 'population', reverse: True }
+        - { sort_key: 'id', reverse: True }
+  - fn: vectordatasource.transform.keep_n_features_gridded
+    params:
+      source_layer: places
+      start_zoom: 11
+      end_zoom: 13
+      items_matching: { kind: locality }
+      max_items: 1
+      grid_width: 12
```

Comment:
I take it all these values for grid width and start/end zoom were arrived at experimentally?  Did you try anything like reducing the width but increasing max_items to get a more organic look?

Reply:
Yep, I did a bit of experimentation based on the suggestions Nathaniel made above and these values seemed to look best. At least in the really dense area around Tokyo.

Reply:
I didn't try more than one item per bucket. That and wider-than-taller buckets (since labels are wider than taller) are two things we could try in the future I think.